### PR TITLE
chore: Initial commit

### PR DESCRIPTION
**Pull request by SlayerOrnstein**
### What did you fix? 
Generate mod images using information from `warframe-items` using node-canvas to generate the final image

---

### Reproduction steps

---

### Evidence/screenshot/link to line

VitalityArchonIntensifySteel ChargeStreamline
Riven's don't have any image but the structure for it's image is there as a safety net
Archgun Riven Mod

### Considerations
- Does this contain a new dependency? **Yes**
- Does this introduce opinionated data formatting or manual data entry? **Yes**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **No**
- Have I run the linter? **Yes**
- Is is a bug fix, feature request, or enhancement? **Feature**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "mod-generator",
```
Comment (TobiTenno) [inline package.json:2]:
```suggestion
  "name": "@wfcd/mod-generator",
  "public": "true",
```

File: tsconfig.json
Comment (TobiTenno) [inline tsconfig.json:1]: could we at least remove the comments?

**Review by TobiTenno — COMMENTED** [review]
seems cool, let's clean up just a touch, shouldn't need so many `@ts-ignore`'s if we can get https://github.com/WFCD/warframe-items/pull/559 merged

File: tsconfig.json
Comment (SlayerOrnstein) [reply tsconfig.json:1]: Yup just had them there so I can uncomment if I needed something. Was gonna clean it up before merge 😂

**Review by TobiTenno — APPROVED** [review]

File: src/utils.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export const modDescription = (
 ): string | undefined => {
   if (description && description.length !== 0) return description;
 
-  if (levelStats) {
```
Comment (TobiTenno) [inline src/utils.ts]: `if (levelStats?.[rank]) {`

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+};
+
+export const modDescription = (
+  description: string | undefined,
+  levelStats: LevelStat[] | undefined,
+  rank: number
+): string | undefined => {
+  if (description && description.length !== 0) return description;
+
+  if (levelStats && levelStats[rank]) {
```
Comment (TobiTenno) [inline src/utils.ts:117]:
```suggestion
  if (levelStats?.[rank]) {
```

**Pull request merged**